src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -498,16 +533,19 @@ Transactor::checkSingleSign(PreclaimContext const& ctx)
 {
     // Check that the value in the signing key slot is a public key.
     auto const pkSigner = ctx.tx.getSigningPubKey();
-    if (!publicKeyType(makeSlice(pkSigner)))
+    if (!(ctx.flags & tapDRY_RUN) && !publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace())
             << "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH;  // FIXME: should be better error!
     }
 
     // Look up the account.
-    auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
     auto const idAccount = ctx.tx.getAccountID(sfAccount);
+    // This ternary is only needed to handle `simulate`
+    auto const idSigner = pkSigner.empty()
```

Comment:
Please move `idSigner` initialization outside of `idAccount` initialization and `sleAccount` check block, it mess with the logic there

Reply:
Fixed in 0b818f33287094da7e46370c20269dac49b7a540